didman/didman_test.go

```diff
@@ -533,6 +535,81 @@ func TestDidman_GetCompoundServices(t *testing.T) {
 	})
 }
 
+func TestDidman_SearchOrganizations(t *testing.T) {
+	id, _ := did.ParseDID("did:nuts:123")
+	docWithService := did.Document{
+		ID: *id,
+		Service: []did.Service{{
+			Type:            "eOverdracht",
+			ServiceEndpoint: map[string]interface{}{"foo": "http://example.org"},
+		}},
+	}
+	docWithoutService := did.Document{
+		ID: *id,
+	}
+	//
+
+	t.Run("ok - no results", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.vcr.EXPECT().Search("organization", map[string]string{"organization.name": "query"}).Return([]concept.Concept{}, nil)
+
+		actual, err := ctx.instance.SearchOrganizations("query", nil)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, actual)
+		assert.Empty(t, actual)
+	})
+	cpt := map[string]interface{}{
+		"subject":      id.String(),
+		"organization": concept.Concept{},
+	}
+	t.Run("ok - no DID service type", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.vcr.EXPECT().Search("organization", map[string]string{"organization.name": "query"}).Return([]concept.Concept{cpt}, nil)
+		ctx.store.EXPECT().Resolve(*id, nil).Return(&docWithoutService, nil, nil)
+
+		actual, err := ctx.instance.SearchOrganizations("query", nil)
+
+		assert.NoError(t, err)
+		assert.Len(t, actual, 1)
+	})
+	t.Run("ok - with DID service type (matches)", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.vcr.EXPECT().Search("organization", map[string]string{"organization.name": "query"}).Return([]concept.Concept{cpt}, nil)
+		ctx.store.EXPECT().Resolve(*id, nil).Return(&docWithService, nil, nil)
+
+		serviceType := "eOverdracht"
+		actual, err := ctx.instance.SearchOrganizations("query", &serviceType)
+
+		assert.NoError(t, err)
+		assert.Len(t, actual, 1)
+	})
+	t.Run("ok - with DID service type (no match)", func(t *testing.T) {
+		ctx := newMockContext(t)
+		ctx.vcr.EXPECT().Search("organization", map[string]string{"organization.name": "query"}).Return([]concept.Concept{cpt}, nil)
+		ctx.store.EXPECT().Resolve(*id, nil).Return(&docWithoutService, nil, nil)
+
+		serviceType := "eOverdracht"
+		actual, err := ctx.instance.SearchOrganizations("query", &serviceType)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, actual)
+		assert.Empty(t, actual)
+	})
+	t.Run("ok - DID document not found (logs, omits result)", func(t *testing.T) {
```

Comment:
missing tests for deactivated or no active controllers, those would fail.

Reply:
Isn't it a bit odd the differ between a deactivated DID Document and "no active controller" when resolving? We could see a DID document without (active) controllers as "deactivated"? That would make resolver error checking a bit simpler (one less error to check).

Reply:
Another round of PR feedback added that return value. It allows for a distinction to be made to the caller of the API why he can't find a DID.